src/main/java/uk/gov/digital/ho/hocs/document/service/MsgDocumentConverter.java

```diff
@@ -99,39 +107,109 @@ public MsgContents extractContents(InputStream inputStream) throws DocumentExcep
+
         try {
-            ElementList elements = XMLWorkerHelper.parseToElementList(contents.getBodyHTML(), null);
-            if (!elements.isEmpty()) {
-                document.add(new Paragraph(""));
-                for (Element element : elements) {
-                    document.add(element);
+            ArrayList<String> paragraphs = new ArrayList<>(List.of(
+                    String.format(FROM_STRING, contents.getFromEmail(), StringUtils.isEmpty(contents.getFromName()) ? "N/A" : contents.getFromName()),
+                    String.format(TO_STRING, StringUtils.isEmpty(contents.getToEmail()) ? contents.getToName() : contents.getToEmail()),
+                    String.format(SUBJECT_STRING, contents.getSubject().replaceAll("\\P{Print}", "")),
+                    String.format(SENT_ON_STRING, contents.getSentOn())
+            ));
+
+            List<String> bodyParagraphs = Arrays.stream(contents.getBodyText().split("\r\n")).map(p -> p.replaceAll("\\P{Print}", "")).collect(Collectors.toList());
+
+            paragraphs.addAll(bodyParagraphs);
+
+            PDRectangle mediabox = pdf.getPage(0).getMediaBox();
+            float width = mediabox.getWidth() - 2 * margin;
+
+            List<String> lines = new ArrayList<>();
+
+            for (String paragraph : paragraphs) {
+                if (paragraph.length() == 0) {
+                    lines.add("");
+                }
+                int lastSpace = -1;
+                while (paragraph.length() > 0) {
+                    int spaceIndex = paragraph.indexOf(' ', lastSpace + 1);
+                    if (spaceIndex < 0)
+                        spaceIndex = paragraph.length();
+                    String subString = paragraph.substring(0, spaceIndex);
+                    float size = fontSize * font.getStringWidth(subString) / 1000;
+                    if (size > width) {
+                        if (lastSpace < 0)
+                            lastSpace = spaceIndex;
+                        subString = paragraph.substring(0, lastSpace);
+                        lines.add(subString);
+                        paragraph = paragraph.substring(lastSpace).trim();
+                        lastSpace = -1;
+                    } else if (spaceIndex == paragraph.length()) {
+                        lines.add(paragraph);
+                        paragraph = "";
+                    } else {
+                        lastSpace = spaceIndex;
+                    }
                 }
             }
-        } catch (RuntimeWorkerException | DocumentException | IOException e) {
+            return lines;
+        } catch (IOException e) {
             log.warn("Failed to Parse MSG Elements: {}", e.getMessage(), value(EVENT, DOCUMENT_CONVERSION_MSG_PARSE_FAILURE), value(EXCEPTION, e));
-            return false;
+            return List.of();
+        }
+    }
+
+    public void printContents(PDPageContentStream contentStream, List<String> lines, PDDocument pdf) {
+        try {
```

Comment:
Use try with resources here or add a finally block to close the streams to avoid memory leaks on exception